### PR TITLE
Isolate data dumping to file

### PR DESCRIPTION
**Pull request by laszlourszuly**
There is an issue with synching dumped data to file system which
sometimes corrupts our persisted test state. This change aims to
mitigate that issue.

### CODE REVIEW

File: explorer/service/src/main/java/org/radixdlt/explorer/helper/DumpHelper.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                // Ensure parent folder exists
+                path.toAbsolutePath().getParent().toFile().mkdirs();
+                Path tmpFilePath = path.getParent().resolve("tmp.txt").toAbsolutePath();
+                Path targetFilePath = path.toAbsolutePath();
+
+                try (FileOutputStream fileOutputStream = new FileOutputStream(tmpFilePath.toString())) {
+                    fileOutputStream.write(data);
+                    fileOutputStream.flush();
+                    fileOutputStream.getFD().sync();
+                    Files.move(tmpFilePath, targetFilePath, ATOMIC_MOVE, REPLACE_EXISTING);
```
Comment (blenessy) [inline explorer/service/src/main/java/org/radixdlt/explorer/helper/DumpHelper.java]: Can we move the file **after** the `fileOutputStream` has been closed?

**Review by blenessy — CHANGES_REQUESTED** [review]
Would it make sense to add some unit tests for the DumpHelper ?

File: explorer/service/src/main/java/org/radixdlt/explorer/helper/DumpHelper.java (hunk 1 above)
Comment (laszlourszuly) [reply explorer/service/src/main/java/org/radixdlt/explorer/helper/DumpHelper.java]: Very good point! I'll change immediately.

**Comment by laszlourszuly** [comment]
Please have a second look @blenessy. The requested unit tests exposed some minor bugs but the API of the `DumpHelper` has also been reworked a bit to be more convenient to use.

**Comment by swiec** [comment]
Tested on the integration branch. Works fine. metrics.csv persisted, doesn't seem to be corrupted.

**Comment by blenessy** [comment]
@laszlourszuly can we get a rebase on this one please ?

**Pull request merged**